Automattic/wp-calypso
# Handle SaaS products as Marketplace products

A new type of Marketplace product will be handled. The product will redirect to a third-party vendor when purchasing.

#### Proposed Changes

* The SaaS products will not display price information
* When upgrade is required, the SaaS product details will have two buttons, one to redirect to the Third Party vendor for purchasing, and another to directly Upgrade to a Business plan
* When no upgrade is required, there will be only one button that will redirect to the Third Party vendor

|Screen|Upgrade needed | Business or greater|
|-------|-------|------|
|Plugin list|   <img width="532" alt="CleanShot 2022-10-19 at 11 02 10@2x" src="https://user-images.githubusercontent.com/3519124/196647160-1e42df7c-102e-45c8-a001-6fae4f186ae4.png">         |               <img width="532" alt="CleanShot 2022-10-19 at 11 04 43@2x" src="https://user-images.githubusercontent.com/3519124/196647311-60e4ec7a-c5d3-41b0-8032-e2f91482e62d.png">         |
|Plugin details|         ![CleanShot 2022-10-19 at 11 07 36@2x](https://user-images.githubusercontent.com/3519124/196648024-f7e9923d-c7b6-42fd-88f6-6a8b6e39ba94.png)          |              ![CleanShot 2022-10-19 at 11 07 21@2x](https://user-images.githubusercontent.com/3519124/196647988-8d169f2a-d2bb-4f5c-90cd-3f3f97558e88.png)        |


#### Testing Instructions



* Sandbox the `public-api` and make sure you are using the `USE_STORE_SANDBOX ` as explained here PCYsg-IA-p2

###### Upgrade needed
* Navigate to the plugins page in a site with a plan that requires an upgrade to install plugins, for example `/plugin/<premium_site>`
* Search for a plugin named `Test Saas Product`
* Make sure the price is not displayed in the plugins list results of the `Test Saas Product`
* Click the `Test Saas Product` to open the details
* Make sure the billing interval switcher is not displayed
* Make sure there are two buttons as in the screenshot above `Plugin details` + `Upgrade needed`
* Click on the `Get started` button and make sure you are redirected to the configured URL (currently pointing to a test URL)
* Go back and click on the `Upgrade to Business` button. Make sure you are redirected to the checkout process with the Business plan in the cart.

###### Business site
* Navigate to the plugins page in a site with a business plan, for example `/plugin/<business_site>`
* Search for a plugin named `Test Saas Product`
* Make sure the price is not displayed in the plugins list results of the `Test Saas Product`
* Click the `Test Saas Product` to open the details
* Make sure the billing interval switcher is not displayed
* Make sure there is one button with the text `Getting started` as in the screenshot above `Plugin details` + `Business or greater`
* Click on the `Get started` button and make sure you are redirected to the configured URL (currently pointing to a test URL)


###### Regression testing
* On any site
* Navigate to a premium plugin, e.g. `Yoast SEO Premium` -> `/plugins/wordpress-seo-premium/<site_id>`
* Make sure the result is as expected
* Navigate to a free plugin, e.g. `WooCommerce` -> `/plugins/woocommerce/<site_id>`
* Make sure the result is as expected


#### Pre-merge Checklist






Fixes #67657
